Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionActionCocoa.mm

```diff
@@ -808,46 +813,67 @@ - (void)_otherPopoverWillShow:(NSNotification *)notification
     if (!extensionContext())
         return;
 
+    // The popup might have presented already.
+    if (popupPresented() || presentsPopupWhenReady())
```

Comment:
I might be misunderstanding what `presentsPopupWhenReady` means here, but this is a bit confusing. With this naming, it seems like we wouldn't want the early return here for `presentsPopupWhenReady`.

Does that just mean the popup isn't ready to present yet?

Reply:
Correct, it isn't ready yet and it is already queued to present. I'll update this comment.